### ISSUE TITLE
Unhandled Exception in Fody Call - VS2015, Costura 1.6.2, Fody 2.3.2 and 2.0.0

### ISSUE THREAD

**Issue by phrebh** [Fody/Costura#244]
Bear with me because this issue is odd and difficult to troubleshoot.

**Timeline**
- 2 months ago, I created a project using the iTextSharp 5.5.12 package and used Costura.Fody to create one EXE. (Thank you so much. This tool is amazing!)
- 2 weeks ago, I began a new project using the same package. At some point, I added a testing project to the solution, but then removed it.
- Yesterday, I got everything to work, then added the Costura package. I immediately got two warnings: the first one was the Fody exception and the second one was a VS one, code IDE0006, which just references the first one when I checked the logs. I updated the version of Fody, but that just changed the details of the exception.
- 2 hours ago, I recreated the project from scratch, added the Costura package when it was mostly empty, and began importing the pieces one at a time. When I added my form back in, I got the error. Aha, so I thought.
- 1 hour ago, I created a new project. The form is empty. I added the iTextSharp package. I then added the Costura package and the error popped up. I reopened my project from two months ago, thinking I messed up a VS setting, but it loads and compiles without issue.
- As I typed this, I put Fody back to version 2.0.0 in my test project to get the original warning, was forced to restart VS because v2.3.2 didn't uninstall properly, and then the message didn't appear. I started a debug run, it compiled correctly, my empty form opened, I closed it, then I got an error message that the FodyWeavers.xml was missing a weaver. I added the <Costura /> line to the file and it compiled again, but I get two message about missing schema info on Weavers and Costura. 
- After that, I went back to my actual project, downgraded Fody back to 2.0.0, ran the project successfully, got the missing weaver message, added the Costura line, then still got the same exception when I built the project! Removing the <Costura /> line works. I then updated Fody back to 2.3.2. It apparently still works as long as I leave the weaver file empty, except it doesn't save the iTextSharp dll into the final EXE.

**Underlying Issue**
I realize that most of that was probably unnecessary, but I included it all because I can't pinpoint the issue. 

The exception text is:
`Severity	Code	Description	Project	File	Line	Suppression State`
`Warning		Fody: An unhandled exception occurred:`
`Exception:`
`Object reference not set to an instance of an object.`
`StackTrace:`
`   at ModuleWeaver.FindMsCoreReferences() in  C:\projects\costura\src\Costura\MsCoreReferenceFinder.cs:line 13`
`   at ModuleWeaver.Execute() in C:\projects\costura\src\Costura\ModuleWeaver.cs:line 29`
`   at lambda_method(Closure , Object )`
`   at InnerWeaver.ExecuteWeavers()`
`   at InnerWeaver.Execute()`
`Source:`
`Costura.Fody`
`TargetSite:`
`Void FindMsCoreReferences() <PROJECT NAME>`

The build output is:
1>------ Build started: Project: PROJECT NAME, Configuration: Debug Any CPU ------
1>    Fody: Fody (version 1.0.0.0) Executing
1>      Fody/Costura:   	No reference to 'Costura.dll' found. References not modified.
1>MSBUILD : error : Fody: An unhandled exception occurred:
1>MSBUILD : error : Exception:
1>MSBUILD : error : Object reference not set to an instance of an object.
1>MSBUILD : error : StackTrace:
1>MSBUILD : error :    at ModuleWeaver.FindMsCoreReferences() in C:\projects\costura\src\Costura\MsCoreReferenceFinder.cs:line 13
1>MSBUILD : error :    at ModuleWeaver.Execute() in C:\projects\costura\src\Costura\ModuleWeaver.cs:line 29
1>MSBUILD : error :    at lambda_method(Closure , Object )
1>MSBUILD : error :    at InnerWeaver.ExecuteWeavers()
1>MSBUILD : error :    at InnerWeaver.Execute()
1>MSBUILD : error : Source:
1>MSBUILD : error : Costura.Fody
1>MSBUILD : error : TargetSite:
1>MSBUILD : error : Void FindMsCoreReferences()
1>MSBUILD : error : 
1>    Fody:   Finished Fody 501ms.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

I've concluded that it's something stupid I did with my project, specifically with my form. I debated on even submitting this, but I'm hoping that if I can't figure out what I did, someone else will be able to point it out. In the meantime, I am going to recreate my form from scratch and hope for the best.

**Comment by phrebh** [comment]
Replacing the form didn't work. I noticed that the tasks folder of Fody 2.3.2 specifies net46, and I'm working on 4.5.2, so I downgraded to Fody 2.0.0 and added <Costura /> to the weavers file.

The build output is:
1>------ Build started: Project: PROJECT NAME, Configuration: Debug Any CPU ------
1>    Fody: Fody (version 2.0.0.0) Executing
1>      Fody/Costura:   	No reference to 'Costura.dll' found. References not modified.
1>      Fody/Costura:   	Embedding 'C:\Users\path\packages\iTextSharp.5.5.12\lib\itextsharp.dll'
1>MSBUILD : error : Fody: An unhandled exception occurred:
1>MSBUILD : error : Exception:
1>MSBUILD : error : Length cannot be less than zero.
1>MSBUILD : error : Parameter name: length
1>MSBUILD : error : StackTrace:
1>MSBUILD : error :    at System.String.Substring(Int32 startIndex, Int32 length)
1>MSBUILD : error :    at Mono.Cecil.Pdb.NativePdbReader.GetImport(PdbScope scope, ModuleDefinition module)
1>MSBUILD : error :    at Mono.Cecil.Pdb.NativePdbReader.ReadScopeAndLocals(PdbScope scope, MethodDebugInformation info)
1>MSBUILD : error :    at Mono.Cecil.Pdb.NativePdbReader.Read(MethodDefinition method)
1>MSBUILD : error :    at Mono.Cecil.Cil.CodeReader.ReadMethodBody()
1>MSBUILD : error :    at Mono.Cecil.Cil.CodeReader.ReadMethodBody(MethodDefinition method)
1>MSBUILD : error :    at Mono.Cecil.MethodDefinition.<>c.<get_Body>b__41_0(MethodDefinition method, MetadataReader reader)
1>MSBUILD : error :    at Mono.Cecil.ModuleDefinition.Read[TItem,TRet](TRet& variable, TItem item, Func`3 read)
1>MSBUILD : error :    at Mono.Cecil.MethodDefinition.get_Body()
1>MSBUILD : error :    at ModuleWeaver.FindInitializeCalls() in C:\projects\costura\src\Costura\CallAttach.cs:line 35
1>MSBUILD : error :    at ModuleWeaver.CallAttach(Configuration config) in C:\projects\costura\src\Costura\CallAttach.cs:line 9
1>MSBUILD : error :    at ModuleWeaver.Execute() in C:\projects\costura\src\Costura\ModuleWeaver.cs:line 37
1>MSBUILD : error :    at lambda_method(Closure , Object )
1>MSBUILD : error :    at InnerWeaver.ExecuteWeavers() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 182
1>MSBUILD : error :    at InnerWeaver.Execute() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 83
1>MSBUILD : error : Source:
1>MSBUILD : error : mscorlib
1>MSBUILD : error : TargetSite:
1>MSBUILD : error : System.String Substring(Int32, Int32)
1>MSBUILD : error : 
1>    Fody:   Finished Fody 3822ms.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

**Comment by phrebh** [comment]
I just updated Fody in the project from 2 months ago and got the same error, which tells me that Costura isn't playing nicely with Fody 2.3.3 (which came out between my first post and this comment).

**Comment by phrebh** [comment]
Through a series of odd tests, I've determined that the problem was the name of my project. The errors pop up if I use "Amortization Schedule Creator" in the Assembly name or "Amortization_Schedule_Creator" in the Root namespace. I changed the assembly to "Amortization Creator" and the root to "Amortization_Creator", but still got the error. I also tried removing the root namespace altogether (based on a MS doc) and it worked at first, but not subsequently. I finally changed the root to "ABC_Amort" and it worked.

So, I've got a workaround, but I certainly didn't expect to need it.
